### PR TITLE
added min size restriction for Text canvas

### PR DESCRIPTION
**Pull request by dolow**
Fix for this issue,  details described in there.
https://github.com/pixijs/pixi.js/issues/4618

This pull request also contains unit test to ensure minimum canvas size.

### CODE REVIEW

**Comment by cursedcoder** [comment]
Straight copy-paste from @bigtimebuddy code, seems legit

File: src/core/text/Text.js [hunk 1]
```diff
@@ -144,8 +144,8 @@ export default class Text extends Sprite
         const maxLineWidth = measured.maxLineWidth;
         const fontProperties = measured.fontProperties;
 
-        this.canvas.width = Math.ceil((width + (style.padding * 2)) * this.resolution);
-        this.canvas.height = Math.ceil((height + (style.padding * 2)) * this.resolution);
+        this.canvas.width = Math.ceil((Math.max(1, width) + (style.padding * 2)) * this.resolution);
```
Comment (themoonrat) [inline src/core/text/Text.js]: wouldn't you want the Math.max to apply over the _entire_ calculation? Because as it stands right now, if I had a resolution multiplier of less than 1, I could get a canvas width of less than 1 again

Comment (cursedcoder) [reply src/core/text/Text.js]: don't think you ever want to pass negative padding, and here `width` is copied from `measuredWidth` which is `0` because of valid symbol `\n`

Comment (bigtimebuddy) [reply src/core/text/Text.js]: But the `ceil` would round that fraction up to 1... `width` = 1.

```js
Math.ceil ((1 + 0) * 0.5);
```
That would equal 1.

Comment (themoonrat) [reply src/core/text/Text.js]: Yeah, true.
Just seems odd, that when the goal is to make sure that the width is at least 1, it's done on part of a calculation rather than the whole thing, and relying on a ceil of a max  *shrugs*. Just could read a lot simpler.
I'll remove my request to change since others are happy tho

**Comment by bigtimebuddy** [comment]
Thanks @dolow!
